piccolo/testing/random_builder.py

```diff
@@ -1,12 +1,50 @@
+from __future__ import annotations
+
 import datetime
 import enum
 import random
 import string
 import typing as t
 import uuid
+from decimal import Decimal
+from functools import partial
+from uuid import UUID
+
+from piccolo.columns import Array, Column
 
 
 class RandomBuilder:
+    @classmethod
+    def _build(cls, column: Column) -> t.Any:
+        if e := column._meta.choices:
+            return cls.next_enum(e)
+
+        mapper: t.Dict[t.Type, t.Callable] = {
+            bool: cls.next_bool,
+            bytes: cls.next_bytes,
+            datetime.date: cls.next_date,
+            datetime.datetime: partial(
+                cls.next_datetime, getattr(column, "tz_aware", False)
+            ),
+            float: cls.next_float,
+            Decimal: partial(
+                cls.next_decimal, column._meta.params.get("digits")
+            ),
+            int: cls.next_int,
+            str: partial(cls.next_str, column._meta.params.get("length")),
+            datetime.time: cls.next_time,
+            datetime.timedelta: cls.next_timedelta,
+            UUID: cls.next_uuid,
+        }
+
+        random_value_callable = mapper.get(column.value_type)
+        if random_value_callable is None:
+            random_value_callable = partial(
+                cls.next_list,
+                mapper[t.cast(Array, column).base_column.value_type],
+            )
```

Comment:
The handling of `list` is quite complex, involving two aspects: the simple callable, such as `RandomBuilder.next_bool`, and additional logic that requires knowledge of the `column` to manufacture another callable. Since `list` relies on obtaining the type from the `mapper`, we cannot determine the exact type until the last moment.

Reply:
You're right - arrays are always tricky edge cases.

Maybe we could add a check if `isinstance(column, Array)`?

Reply:
See my latest comment regarding this issue.